front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/Export.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { withRouter, WithRouterProps } from 'react-router';
+import { API_PATH } from 'containers/App/constants';
+
+// components
+import Button from 'components/UI/Button';
+
+// intl
+import { injectIntl } from 'utils/cl-intl';
+import messages from '../../messages';
+import { InjectedIntlProps } from 'react-intl';
+
+// utils
+import { colors } from 'utils/styleUtils';
+import { requestBlob } from 'utils/request';
+import { isNilOrError } from 'utils/helperUtils';
+
+// hooks
+import useInsightsView from 'modules/commercial/insights/hooks/useInsightsView';
+
+// services
+import { getInsightsInputsEndpoint } from 'modules/commercial/insights/services/insightsInputs';
+
+type ExportProps = WithRouterProps & InjectedIntlProps;
+
+const Export = ({
+  params: { viewId },
+  location: { query },
+  intl: { formatMessage, formatDate },
+}: ExportProps) => {
+  const view = useInsightsView(viewId);
+
+  if (isNilOrError(view)) {
+    return null;
+  }
+
+  const handleExportClick = async () => {
+    try {
+      const blob = await requestBlob(
+        `${API_PATH}/${getInsightsInputsEndpoint(viewId)}/as_xlsx`,
+        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+        {
+          category: query.category,
+          processed:
+            query.processed === 'true'
+              ? true
+              : query.processed === 'false'
+              ? false
+              : undefined,
+        }
+      );
+      saveAs(
+        blob,
+        `${formatMessage(messages.inputsTableExportFileName)}_${
+          view.attributes.name
+        }_${formatDate(Date.now())}.xlsx`
+      );
+    } catch {
+      // Do nothing
+    }
+  };
+
+  return (
+    <div data-testid="insightsExport">
```

Comment:
Can't you add the data-testid to the Button directly?

Reply:
Not yet, the Button takes and passes down specific props and data attributes are not among them. It will be possible once we add the Box wrapper to the component in the component library though. I'm working on it! ;)